### PR TITLE
Adds Formam as a replacement for Schema

### PR DESCRIPTION
**Pull request by apaganobeleno**
This is related with #335

### CODE REVIEW

File: binding.go [hunk 1]
```diff
@@ -28,23 +31,51 @@ func RegisterBinder(contentType string, fn BinderFunc) {
 	binders[contentType] = fn
 }
 
-func init() {
-	schemaDecoder = schema.NewDecoder()
-	schemaDecoder.IgnoreUnknownKeys(true)
-	schemaDecoder.ZeroEmpty(true)
+// RegisterTimeLayout allows to add custom time layouts that
+// the binder will be able to use for decoding.
+func RegisterTimeLayout(layout string) {
```
Comment (markbates) [inline binding.go]: Instead of `Layout` lets use the term `Format`

File: binding.go [hunk 2]
```diff
@@ -28,23 +31,51 @@ func RegisterBinder(contentType string, fn BinderFunc) {
...
+	timeLayouts = append(timeLayouts, layout)
+}
 
-	// register the types in the nulls package with the decoder
-	nulls.RegisterWithSchema(func(i interface{}, fn func(s string) reflect.Value) {
-		schemaDecoder.RegisterConverter(i, fn)
+// RegisterBinderTypeDecoder allows to define custom type decoders.
+func RegisterBinderTypeDecoder(fn formam.DecodeCustomTypeFunc, types []interface{}, fields []interface{}) {
```
Comment (markbates) [inline binding.go]: let's not exposed `formam` directly. Let's expose our own "type" that maps to theirs.

File: binding.go [hunk 3]
```diff
@@ -28,23 +31,51 @@ func RegisterBinder(contentType string, fn BinderFunc) {
...
+}
+
+func init() {
+	decoder = formam.NewDecoder(&formam.DecoderOptions{
+		TagName:           "schema",
+		IgnoreUnknownKeys: true,
 	})
 
-	sb := func(req *http.Request, value interface{}) error {
+	decoder.RegisterCustomType(func(vals []string) (interface{}, error) {
```
Comment (markbates) [inline binding.go]: let's register a similar function for `nulls.Time`

**Comment by apaganobeleno** [comment]
Sounds good, i'll work on these changes as well as the new package, thanks for the feedback.

File: binding.go [hunk 4]
```diff
@@ -28,23 +31,51 @@ func RegisterBinder(contentType string, fn BinderFunc) {
...
-	nulls.RegisterWithSchema(func(i interface{}, fn func(s string) reflect.Value) {
-		schemaDecoder.RegisterConverter(i, fn)
+// RegisterBinderTypeDecoder allows to define custom type decoders.
+func RegisterBinderTypeDecoder(fn formam.DecodeCustomTypeFunc, types []interface{}, fields []interface{}) {
+	decoder.RegisterCustomType(fn, types, fields)
+}
+
+func init() {
+	decoder = formam.NewDecoder(&formam.DecoderOptions{
+		TagName:           "schema",
```
Comment (markbates) [inline binding.go]: Let's set the tag to `form` instead. See this ticket https://github.com/gobuffalo/tags/issues/26

File: binding.go (hunk 1 above)
Comment (apaganobeleno) [inline binding.go]: 👍

File: binding.go (hunk 4 above)
Comment (apaganobeleno) [inline binding.go]: 👍

File: binding.go (hunk 2 above)
Comment (apaganobeleno) [inline binding.go]: 👍

**Comment by markbates** [comment]
@apaganobeleno I made a few changes, can you review and tell me what you think?

**Comment by apaganobeleno** [comment]
@markbates looks very elegant Sir! thanks for those changes.

**Pull request merged**